### ISSUE TITLE
URL environment variable not working

### ISSUE THREAD

**Issue by felixge** [jenkinsci/hipchat-plugin#93]
In my Jenkinsfile, I'm trying to send a hipchat message like this:

```
  hipchatSend (
    color: 'PURPLE',
    message: "STARTED: ${URL}"
  )
```

However, in HipChat I see the following being displayed:

```
STARTED: class java.net.URL
```

Workaround: Using `${env.BUILD_URL}` seems to work, but that doesn't get me the blue ocean URLs I was hoping to get.

Many thanks in advance for taking a look!

**Comment by aldaris** [comment]
Thanks for the report, will try to reproduce it, but may need some extra info a bit later.

**Comment by felixge** [comment]
@aldaris thanks. Just let me know what info you need!

**Comment by aldaris** [comment]
I was able to locally reproduce this.
When debugging this, it turned out that the message passed to the build step implementation already had the $URL variable resolved. This was most likely done by the pipeline system in my opinion. Changing the DSL to:

```
hipchatSend color: 'YELLOW', message: "STARTED: $URL"
```

helped, the single quotes made sure that variables weren't resolved by the pipeline system (or Groovy), and the resolution was left to the HipChat plugin instead. Closing this as not a defect.
